### PR TITLE
Dependencies | September 5th 2022

### PR DESCRIPTION
**Pull request by coldlink**
## What does this change?

Changes from previous #1918 :
- Brings our dependencies up to the latest versions where possible

Holds back the following:
* Keeps `react`, `react-dom` and `react-test-renderer` at v17 because v18 will break IE11 support going forward
* Keeps `@types/react` and `@types/react-dom` at v17 as v18 depends on React 18
* Keeps `@testing-library/react` at v12 because v13 depends on React 18
* Keeps `@testing-library/react-hooks` at v7 because v8 depends on React 18
* Keeps `@types/react-test-renderer` and `react-test-renderer` at v17 because v18 depends on React 18
* Holds back upgrading `@guardian/` components as they're major version updates which broke the build for some reason

## Tested

- [x] CODE
- [x] IE11

### CODE REVIEW

File: src/server/lib/__tests__/headers.test.ts [hunk 1]
```diff
@@ -108,7 +108,6 @@ describe('Content Security Policy headers', () => {
           'o14302.ingest.sentry.io',
         ].join(' '),
       );
-      expect(splitCSPHeader).toContain('block-all-mixed-content');
```
Comment (kelvin-chappell) [inline src/server/lib/__tests__/headers.test.ts:111]: Is this no longer useful or is it no longer being generated?

**Review by kelvin-chappell — APPROVED** [review]
👍

File: src/server/lib/__tests__/headers.test.ts (hunk 1 above)
Comment (coldlink) [reply src/server/lib/__tests__/headers.test.ts:111]: its no longer being generated by helmet as it's been deprecated https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/block-all-mixed-content

Comment (kelvin-chappell) [reply src/server/lib/__tests__/headers.test.ts:111]: Thanks!

**Pull request merged**